# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [10]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [11]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6        V7       
1 0.6768432 0.9279591 0.4536549 0.9529238 0.3405527 0.6417581 0.3087642
2 0.7993542 0.3730730 0.3406247 0.9641088 0.4147309 0.6028798 0.3799041
  V8        V9        V10       V11        V12       V13       V14      
1 0.3233341 0.8540420 0.1224485 0.91943812 0.7047666 0.7107954 0.3124824
2 0.2417699 0.5015066 0.6433366 0.09565883 0.2829162 0.7368645 0.7644373
  V15       V16       V17        V18       V19       V20      
1 0.5032247 0.7582725 0.02088268 0.6112689 0.5278718 0.9868865
2 0.9678720 0.3331226 0.66427606 0.8737952 0.6105275 0.7235095

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [12]:
df2 <- gather(df1)
  

head(df2)

key value    
1 V1  0.6768432
2 V1  0.7993542
3 V2  0.9279591
4 V2  0.3730730
5 V3  0.4536549
6 V3  0.3406247

## Assignment 2

We will generate some data:

In [13]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [14]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [18]:
df5 <- full_join(df3,df4,by="id")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook